cmd/gateway-gcs.go

```diff
@@ -951,8 +952,8 @@ func (l *gcsGateway) CompleteMultipartUpload(bucket string, key string, uploadID
 		composeParts := make([]*storage.ObjectHandle, composeCount)
 		for i := 0; i < composeCount; i++ {
 			// Create 'composed-object-N' using next 32 parts.
-			composeName := fmt.Sprintf("composed-object-%d", i)
-			composeParts[i] = l.client.Bucket(bucket).Object(gcsMultipartDataName(uploadID, composeName))
+			composeObjectName := fmt.Sprintf("%s/%s/composed-object-%05d", gcsMinioMultipartPathV1, uploadID, i)
```

Comment:
better keep this in `minio.sys.temp/tmp/uploadID/composed-object-%05d`? That way backend will be cleaner.
In case we need to implement ListObjectParts in future it will also be easy as we need not have to filter out composed objects.